### PR TITLE
Refactor node run

### PR DESCRIPTION
**Pull request by zhangchiqing**

### CODE REVIEW

File: cmd/node.go [hunk 1]
```diff
@@ -50,28 +80,46 @@ func (node *FlowNodeImp) Run() {
...
+		cancel() // prevent context leak
+		return fmt.Errorf("unhandled irrecoverable error: %w", err)
 	}
 
-	node.Logger.Info().Msgf("%s node shutting down", node.BaseConfig.NodeRole)
+	// if termination signal is received, notify all components to stop gracefully
+	node.Logger.Info().Msgf("received termination signal, graceful shutting down %s node", node.BaseConfig.NodeRole)
 	cancel()
 
-	sigCtx, _ = util.WithSignal(context.Background(), signalChan)
```
[the remark this answers is not in the document]
Comment (zhangchiqing) [reply cmd/node.go:65]: Good catch

[Handling these channel directly here](https://github.com/onflow/flow-go/pull/1606/files#diff-2aa1861c9c209b714de8031f3f5a7fd114330ab28346133c051e22a75c8c9eabR107-L71)

**Review by AlexHentschel — APPROVED** [review]
LGTM (disclaimer: I only gave the PR a high-level review)

**Comment by peterargue** [comment]
Thanks for these changes @zhangchiqing. I did some refactoring that simplified the signalling a bit, and also incorporated a bunch of these changes. Let me know what you think: https://github.com/onflow/flow-go/blob/7f1d3993a79b996d42fa3391670cd5f8b459afaa/cmd/node.go

**Comment by zhangchiqing** [comment]
close for https://github.com/onflow/flow-go/pull/1559#pullrequestreview-824629595

**Pull request closed without merging**